strax/storage/common.py

```diff
@@ -25,130 +27,259 @@ def __repr__(self):
 
         If take_only and exclude are both omitted, provide all data types.
         If a data type is listed in both, it will not be provided.
+        Attempting to read/write unwanted data types throws DataTypeNotWanted.
+        """
+        if overwrite not in 'never if_broken always'.split():
+            raise RuntimeError(f"Invalid 'overwrite' setting {overwrite}. ")
 
-        Attempting to read unwanted data types throws NotCached.
-        Attempting to save unwanted data types throws RuntimeError
-            (you're supposed to check this with the .provides method).
-    """
-
-    def __init__(self,
-                 take_only=tuple(), exclude=tuple(),
-                 recover=False,
-                 readonly=False):
-        self._take_only = strax.to_str_tuple(take_only)
-        self._exclude = strax.to_str_tuple(exclude)
-        self.recover = recover
+        self.take_only = strax.to_str_tuple(take_only)
+        self.exclude = strax.to_str_tuple(exclude)
+        self.overwrite = overwrite
         self.readonly = readonly
         self.log = logging.getLogger(self.__class__.__name__)
 
-    def provides(self, data_type, write=False):
-        """Return whether this store can provides this datatype
-        :param write: if True, return if we can also store the data (in case
-        we do not have it yet)
+    def loader(self, key: DataKey, ambiguous='warn',
+               ignore_lineage=tuple(), ignore_config=tuple(),
+               executor=None):
+        """Return loader for data described by DataKey.
+        :param key: DataKey describing data
+        :param ambiguous: Behaviour if multiple matching data entries are
+        found:
+        - 'error': Raise AmbigousDataRequest exception.
+        - 'warn' (default): warn with AmbiguousDataDescription.
+        - 'ignore': do nothing. Return first match.
+        In the latter two cases, the first match is returned.
+        :param ignore_lineage: list/tuple of plugin names for which no
+        plugin name, version, or option check is performed.
+        :param ignore_config: list/tuple of configuration options for which no
+        check is performed.
+        :param executor: Executor for pushing load computation to
+        """
+        backend, backend_key = self.find(key,
+                                         ambiguous=ambiguous,
+                                         write=False,
+                                         ignore_lineage=ignore_lineage,
+                                         ignore_config=ignore_config)
+        return self._get_backend(backend).loader(backend_key, executor)
+
+    def saver(self, key, metadata, meta_only):
+        """Return saver for data described by DataKey."""
+        backend, backend_key = self.find(key, write=True)
+        return self._get_backend(backend).saver(backend_key,
+                                                metadata, meta_only)
+
+    def get_metadata(self, key,
```

Comment:
What type of metadata is there?  Is this explained somewhere?

Reply:
Ah, this is datatype metadata and NOT run metadata.  Can we clarify?  Also, datatype means table, which is slightly confusing.